src/Compilers/VisualBasic/Portable/Symbols/Source/SourceMemberContainerTypeSymbol.vb

```diff
@@ -3203,19 +3227,83 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
             End Get
         End Property
 
-        Friend Function CalculateLocalSyntaxOffsetInSynthesizedConstructor(localPosition As Integer, localTree As SyntaxTree, isShared As Boolean) As Integer
-            Dim aggregateLength As Integer = 0
-
+        Friend Function CalculateSyntaxOffsetInSynthesizedConstructor(position As Integer, tree As SyntaxTree, isShared As Boolean) As Integer
             If IsScriptClass AndAlso Not isShared Then
+                Dim aggregateLength As Integer = 0
+
                 For Each declaration In Me.m_declaration.Declarations
                     Dim syntaxRef = declaration.SyntaxReference
 
-                    If localTree Is syntaxRef.SyntaxTree Then
-                        Return aggregateLength + localPosition
+                    If tree Is syntaxRef.SyntaxTree Then
+                        Return aggregateLength + position
                     End If
 
                     aggregateLength += syntaxRef.Span.Length
                 Next
+
+                Throw ExceptionUtilities.Unreachable
```

Comment:
It looks like the function didn't throw before, but was returning -1. Consider doing the same and maybe adding an assert instead.